src/services/RetryCountExceededException.js

```diff
@@ -0,0 +1,15 @@
+export default function RetryCountExceededException(requestUnit) {
+  this.message = 'Retry count has been exceeded';
+  this.requestUnit = requestUnit;
+
+  // Use V8's native method if available, otherwise fallback
+  if ("captureStackTrace" in Error) {
+    Error.captureStackTrace(this, RetryCountExceededException);
+  } else {
+    this.stack = (new Error()).stack;
+  }
+}
+
+RetryCountExceededException.prototype = Object.create(Error.prototype);
+RetryCountExceededException.prototype.name = "RetryCountExceededException";
+RetryCountExceededException.prototype.constructor = RetryCountExceededException;
```

Comment:
Babel 6.x does not correctly support extending core classes yet:
http://stackoverflow.com/questions/31089801/extending-error-in-javascript-with-es6-syntax 
https://github.com/babel/babel/issues/3083

there's a babel plugin https://www.npmjs.com/package/babel-plugin-transform-builtin-extend that enables extending from Error and Array. Don't know how we feel about this..